packages/cli/src/ui/components/shared/DescriptiveRadioButtonSelect.tsx

```diff
@@ -0,0 +1,164 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import type React from 'react';
+import { useState, useRef } from 'react';
+import { Text, Box } from 'ink';
+import { theme } from '../../semantic-colors.js';
+import { useKeypress } from '../../hooks/useKeypress.js';
+
+const NUMBER_INPUT_TIMEOUT_MS = 1000;
+
+export interface DescriptiveRadioSelectItem<T> {
+  value: T;
+  title: string;
+  description: string;
+  disabled?: boolean;
+}
+
+export interface DescriptiveRadioButtonSelectProps<T> {
```

Comment:
please refactor this to use composition and a common base class with the existing RadioButton class. There is a lot of duplicate logic and we will make future enhancements that should be adopted by both. Generally this is just a radio button class with nicer rendering of the inputs.